test/jdk/com/sun/java/swing/plaf/gtk/TestBackSpaceAction.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8078471
+ * @key headful
+ * @requires (os.family == "linux")
+ * @summary Verifies if filechooser current directory changed to parent directory on
+ * BACKSPACE key press from keyboard except root directory.
+ * @run main TestBackSpaceAction
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Component;
+import java.awt.event.HierarchyEvent;
+import java.awt.event.HierarchyListener;
+import java.awt.event.KeyEvent;
+import java.awt.Robot;
+import java.io.File;
+import javax.swing.JFileChooser;
+import javax.swing.JPanel;
+import javax.swing.JComboBox;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+
+
+public class TestBackSpaceAction {
+    private static JFrame frame;
+    private static JFileChooser fileChooser;
+    private static Robot robot;
+    private static File testDir;
+    private static File subDir;
+    private static File prevDir;
+    private static File crntDir;
+
+    public static void main(String[] args) throws Exception {
+        robot = new Robot();
+        robot.setAutoDelay(100);
+        try {
+            // create test directory
+            String tmpDir = System.getProperty("java.io.tmpdir");
+            if (tmpDir.length() == 0) { //'java.io.tmpdir' isn't guaranteed to be defined
+                tmpDir = System.getProperty("user.home");
+            }
+            testDir = new File(tmpDir, "testDir");
+            testDir.mkdir();
+            testDir.deleteOnExit();
+
+            // create sub directory inside test directory
+            subDir = new File(testDir, "subDir");
+            subDir.mkdir();
+            subDir.deleteOnExit();
```

Comment:
To my surprise, this should be OK. The spec says
"Files (or directories) are deleted in the reverse order that they are registered."